base/fill/error_collector.go

```diff
@@ -0,0 +1,57 @@
+package fill
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ErrorCollector collects errors, merging them into one, while noting the path where they happen
+type ErrorCollector struct {
```

Comment:
Fun! It's worth calling out that a similar pattern would be applicable to our jsonschema package, which returns a slice of `ValError`: https://github.com/qri-io/jsonschema/blob/master/schema.go#L185